testkit-java/src/main/scala/com/akkaserverless/javasdk/testkit/impl/ActionResultImpl.scala

```diff
@@ -48,13 +51,11 @@ final class ActionResultImpl[T](effect: Action.Effect[T]) extends ActionResult[T
   /** @return true if the call was async, false if not */
   def isAsync(): Boolean = effect.isInstanceOf[ActionEffectImpl.AsyncEffect[T]]
 
-  def getAsyncEffect(): CompletionStage[Action.Effect[T]] = {
+  def getAsyncEffect(): CompletionStage[ActionResultImpl[T]] = {
```

Comment:
No need to be specific here, with `CompletionStage<ActionResult<T>>` above this should be `CompletionStage[ActionResult[T]]` as well.

Reply:
You're right. I need the covariant version, though `  def getAsyncEffect(): CompletionStage[_ <: ActionResult[T]]`.

Reply:
Make the return type invariant and make sure the returned value aligns (we know it does), a type ascription inside map (to `ActionResult[T]`), an explicit type param to `toJava[ActionResult[T]]` or casting the entire before returning result should do the trick.

Reply:
That sounds better than my approach. Nice!

Reply:
done